File: app/src/main/java/ch/zuehlke/sbb/reddit/features/detail/DetailAdapter.java
package ch.zuehlke.sbb.reddit.features.detail;

import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.util.TypedValue;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import java.util.ArrayList;
import java.util.List;

import ch.zuehlke.sbb.reddit.R;
import ch.zuehlke.sbb.reddit.data.source.remote.model.posts.RedditPost;
import ch.zuehlke.sbb.reddit.models.RedditPostsData;
import ch.zuehlke.sbb.reddit.util.DateUtils;

/**
 * Created by chsc on 13.11.17.
 */

public class DetailAdapter extends RecyclerView.Adapter<DetailAdapter.DetailViewHolder>{


    private List<RedditPostsData> mItems = new ArrayList<>();
    private Context mContext;
    public DetailAdapter(Context context){
        mContext = context;
    }

    @Override
    public DetailViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_detail, parent, false);
        return new DetailViewHolder(view);
    }

    @Override
    public void onBindViewHolder(DetailViewHolder holder, int position) {
        RedditPostsData post = mItems.get(position);

        holder.mUsername.setText(post.author);
        holder.mText.setText(post.body);
        holder.mCreated.setText(DateUtils.friendlyTime(post.createdUtc));
        setDepthPadding(holder.itemView,post.depth);
    }

    public void clearAndAddPosts(List<RedditPostsData> newsData){
        int previousItemSize = mItems.size();
        mItems.clear();
        notifyItemRangeRemoved(0,previousItemSize);
        mItems.addAll(newsData);
        notifyDataSetChanged();

    }

    @Override
    public int getItemCount() {
        return mItems.size();
    }

    private void setDepthPadding(View view, int depth){
        float normalPadding = 5f;
        float leftPadding= normalPadding + (depth * 10f);

        int normalDp = (int) TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, normalPadding, mContext.getResources().getDisplayMetrics());
        int leftDp = (int) TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, leftPadding, mContext.getResources().getDisplayMetrics());
        view.setPadding(leftDp, normalDp, normalDp, normalDp);
    }

    static class DetailViewHolder extends RecyclerView.ViewHolder{

        private TextView mUsername;
        private TextView mCreated;
        private TextView mText;

        public DetailViewHolder(View itemView) {
            super(itemView);
            mUsername = itemView.findViewById(R.id.username);
            mCreated = itemView.findViewById(R.id.created);
            mText = itemView.findViewById(R.id.text);
        }
    }
}

File: app/src/main/java/ch/zuehlke/sbb/reddit/features/news/detail/DetailAdapter.kt
package ch.zuehlke.sbb.reddit.features.news.detail

import android.content.Context
import android.support.v7.widget.RecyclerView
import android.util.TypedValue
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView

import java.util.ArrayList

import ch.zuehlke.sbb.reddit.R
import ch.zuehlke.sbb.reddit.models.RedditPostsData
import ch.zuehlke.sbb.reddit.util.DateUtils

/**
 * Created by chsc on 13.11.17.
 */

class DetailAdapter(private val mContext: Context) : RecyclerView.Adapter<DetailAdapter.DetailViewHolder>() {


    private val mItems = ArrayList<RedditPostsData>()

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): DetailViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.item_detail, parent, false)
        return DetailViewHolder(view)
    }

    override fun onBindViewHolder(holder: DetailViewHolder, position: Int) {
        val post = mItems[position]

        holder.mUsername.text = post.author
        holder.mText.text = post.body
        holder.mCreated.text = DateUtils.friendlyTime(post.createdUtc)
        setDepthPadding(holder.itemView, post.depth)
    }

    fun clearAndAddPosts(newsData: List<RedditPostsData>) {
        val previousItemSize = mItems.size
        mItems.clear()
        notifyItemRangeRemoved(0, previousItemSize)
        mItems.addAll(newsData)
        notifyDataSetChanged()

    }

    override fun getItemCount(): Int {
        return mItems.size
    }

    private fun setDepthPadding(view: View, depth: Int) {
        val normalPadding = 5f
        val leftPadding = normalPadding + depth * 10f

        val normalDp = TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, normalPadding, mContext.resources.displayMetrics).toInt()
        val leftDp = TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, leftPadding, mContext.resources.displayMetrics).toInt()
        view.setPadding(leftDp, normalDp, normalDp, normalDp)
    }

     class DetailViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {

         val mUsername: TextView
         val mCreated: TextView
         val mText: TextView

        init {
            mUsername = itemView.findViewById<TextView>(R.id.username)
            mCreated = itemView.findViewById<TextView>(R.id.created)
            mText = itemView.findViewById<TextView>(R.id.text)
        }
    }
}
